File: Easy/Recursive sequence/recursive-sequence.cpp
//{ Driver Code Starts
//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++

class Solution{
public:
    long long sequence(int n){
        // code here
        long long ans = 0;
        int count = 1, mod = 1e9 + 7;
        for(int i = 1; i<=n; i++) {
            long long num = 1;
            for(int j = 1; j<=i; j++) {
                num = (num * (count))%mod;
                count++;
            }
            ans = (ans + num) % mod;
        }
        
        return ans;
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int N;
        cin>>N;
        
        Solution ob;
        cout<<ob.sequence(N)<<endl;
    }
    return 0;
}
// } Driver Code Ends

File: Easy/Recursive sequence/recursive-sequence.py
#User function Template for python3

class Solution:
    def sequence(self, n):
        # code here
        ans, count, mod = 0, 1, 10**9 + 7
        for i in range(1, n+1):
            num = 1
            for j in range(1, i+1):
                num = (num * count) % mod;
                count += 1
            ans = (ans + num) % mod;
        
        return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        
        ob = Solution()
        print(ob.sequence(N))
# } Driver Code Ends